Model/Api/ShippingMethods.php

```diff
@@ -342,16 +337,70 @@ public function getShippingMethods($cart, $shipping_address)
 
             return $shippingOptionsModel;
         } catch (\Magento\Framework\Webapi\Exception $e) {
-            $this->bugsnag->notifyException($e);
-            $this->sendErrorResponse($e->getCode(), $e->getMessage(), $e->getHttpCode());
+            $this->catchExceptionAndSendError($e, $e->getMessage(), $e->getCode(), $e->getHttpCode());
         } catch (BoltException $e) {
-            $this->bugsnag->notifyException($e);
-            $this->sendErrorResponse($e->getCode(), $e->getMessage(), 422);
+            $this->catchExceptionAndSendError($e, $e->getMessage(), $e->getCode());
         } catch (\Exception $e) {
-            $this->bugsnag->notifyException($e);
+            $this->catchExceptionAndSendError($e, '', 6009, 422);
+        }
+    }
+
+    /**
+     * @param        $e
+     * @param string $msg
+     * @param int    $code
+     * @param int    $httpStatusCode
```

Comment:
no need for this param?

Reply:
it is needed. The `httpStatusCode` parameter is passed here, a bit above for `Webapi\Exception`